docs/src/pages/reference/configuration-reference.md

```diff
@@ -3,22 +3,44 @@ layout: ~/layouts/MainLayout.astro
 title: Configuration Reference
 ---
 
-To configure Astro, add an `astro.config.mjs` file in the root of your project. All settings are optional.
-
-You can view the full configuration API (including information about default configuration) on [GitHub.](https://github.com/withastro/astro/blob/latest/packages/astro/src/%40types/astro.ts)
+To configure Astro, add an `astro.config.mjs` file to the root of your project.
 
 ```js
-// Example: astro.config.mjs
-
-// @type-check enabled!
-// VSCode and other TypeScript-enabled text editors will provide auto-completion,
-// helpful tooltips, and warnings if your exported object is invalid.
-// You can disable this by removing "@ts-check" and `@type` comments below.
-
-// @ts-check
-export default /** @type {import('astro').AstroUserConfig} */ (
-  {
-    // ...
-  }
-);
+export default /** @type {import('astro').AstroUserConfig} */ ({
+  projectRoot: './',
+  public: './public/',
+  dist: './dist/',
+  src: './src/',
+  pages: './pages/',
+  renderers: [
+    '@astrojs/renderer-svelte',
+    '@astrojs/renderer-vue',
+    '@astrojs/renderer-react',
+    '@astrojs/renderer-preact',
+  ],
+});
```

Comment:
we should include a `vite:{ } ` option with a demo of how its written, perhaps applying a plugin,